suit_generator/cmd_cache_create.py

```diff
@@ -6,116 +6,297 @@
+
+        self.cache_data += slot_data
+
+    def close_and_save_cache(self, output_file: str):
+        """
+        Close the cache and save it to the specified output file.
+
+        This function closes the cache by adding the end-of-map byte (0xFF) and saves the cache to the specified output
+        file.
+
+        :param output_file: Path to the output raw SUIT DFU cache file.
+        :type output_file: str
+        """
+        self.cache_data += bytes([0xFF])
+        with open(output_file, "wb") as f:
+            f.write(self.cache_data)
+
+    def merge_single_cache_file(self, cache_input_file: str):
+        """
+        Merge the contents of the provided single cache file into the current cache.
+        :param cache_input_file: Path to the input raw SUIT DFU cache file.
+        """
+
+        with open(cache_input_file, "rb") as f:
+            data = f.read()
+
+        cache_dict = cbor2.loads(data)
+
+        for k in cache_dict.keys():
+            if  k == b'':
+                continue # Empty key means padding - skip
+            self.add_cache_slot(k, cache_dict[k])
+
+
+class CacheFromPayloads:
+    def fill_cache_from_payloads(cache: CachePartition, input: list[str]) -> None:
+        """
+        Process list of input binaries, each associated with a URI, and fill the SUIT DFU cache with the data
+
+        :param cache: CachePartition object to fill with the data
+        :param input: List of input binaries with corresponding URIs, passed in the format <URI>,<INPUT_FILE_PATH>
+        """
+
+        for single_input in input:
+            args = single_input.split(",")
+            if len(args) < 2:
+                raise ValueError("Invalid number of input arguments: " + single_input)
+            uri, input_file = args
+
+            with open(input_file, "rb") as f:
+                data = f.read()
+
+            cache.add_cache_slot(uri, data)
 
-        cache_data += slot_data
+class CacheFromEnvelope:
+    def fill_cache_from_envelope_data(cache: CachePartition,
+                                      envelope_data: bytes,
+                                      omit_payload_regex: str,
+                                      dependency_regex: str) -> bytes:
+        """
+        Fill the cache partition with data from the payloads inside the provided envelope binary data.
+        This function is called recursively for dependency envelopes.
 
-    cache_data += bytes([0xFF])  # Finish the indefinite length map
+        :param cache: CachePartition object to fill with the data
+        :param envelope_data: Binary data of the envelope to extract the payloads from
+        :param omit_payload_regex: Integrated payloads matching the regular expression will not be extracted to the cache
+        :param dependency_regex: Integrated payloads matching the regular expression will be treated as dependency envelopes
+        """
+        envelope = cbor2.loads(envelope_data)
```

Comment:
Data which is not a valid suit envelope can potentialy crash during cbor decoding or (if decoded as non taged structure) during access to envelope.value in line 224. 

Example for nordic_top envelope with `--dependency-regex ".*sys.*"`:
```
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x90 in position 0: invalid start byte
```
Example for nordic_top envelope with `--dependency-regex ".*sec.*"`:
```
AttributeError: 'int' object has no attribute 'value'
```

Reply:
Modified error logging so that this results in a hopefully clearer error